src/DotVVM.Framework/Compilation/Binding/ExpressionHelper.cs

```diff
@@ -252,6 +259,53 @@ private static MethodRecognitionResult TryCallMethod(MethodInfo method, Type[] t
             };
         }
 
+        private static Type ResolveGenericTypeFromGivenExpressions(int i, Type genericArgument, ParameterInfo[] parameters, Expression[] args)
+        {
+            for (var j = 0; j < parameters.Length; j++)
+            {
+                var parameter = parameters[j];
+                if (parameter.ParameterType.IsGenericParameter && parameter.ParameterType.GenericParameterPosition == i)
```

Comment:
Why not `parameter.ParameterType == genericArgument`? (I haven't checked if that works, so sorry if it doesn't)

Reply:
I had stuck to the solution that was used before my intervention. I have added several tests and can confirm that this comparison works.